app/touchnet.js

```diff
@@ -42,7 +42,7 @@ class TouchnetWS {
     let response = await touchnetRequest(this.uri, this.auth, authorizeBody(session_id));
     return {
       receipt:  getSingleNode('/soapenv:Envelope/soapenv:Body/tn:authorizeAccountResponse/tn:creditResponse/tn:receiptNumber', response),
-      user_id:  getSingleNode('/soapenv:Envelope/soapenv:Body/tn:authorizeAccountResponse/tn:ticketName', response),
+      user_id:  getSingleNode('/soapenv:Envelope/soapenv:Body/tn:authorizeAccountResponse/tn:ticketName', response).replace(/_\d+$/, ''),
```

Comment:
I think this is the way to go probably--no (purely hypothetical) regex DoS vector--but I need to test it.

```suggestion
    let response = await touchnetRequest(this.uri, this.auth, authorizeBody(session_id));
    const ticket_name = getSingleNode('/soapenv:Envelope/soapenv:Body/tn:authorizeAccountResponse/tn:ticketName', response);
    const user_id = ticket_name.slice(0, ticket_name.lastIndexOf('_'));
     return {
       receipt:  getSingleNode('/soapenv:Envelope/soapenv:Body/tn:authorizeAccountResponse/tn:creditResponse/tn:receiptNumber', response),
      user_id: user_id,
```